securesign/secure-sign-operator
# feat: allow for configurable ctlog private key

This pr allows for configurable ctlog private key

## Testing
1.  Usual, cluster, login, build, deploy
2.  Create a ns for the crd `oc create ns openshift-trusted-artifact-signer`
3. Follow instructions at https://github.com/google/certificate-transparency-go/blob/master/trillian/docs/ManualDeployment.md#key-generation to create a private key.
4. Create secret in ns `oc -n openshift-trusted-artifact-signer create secret generic <secretName> --from-file=<secretKey>=<path/to/private/key.pem>`
5. Update yaml:
```
  ctlog:
    certificate:
      create: false
      secretName: <secretName>
      secretKey: <secretKey>
```
6. Apply yaml file
7. Watch for errors and sign and verify

Completes https://github.com/securesign/secure-sign-operator/issues/33
